src/bucket.ts

```diff
@@ -30,11 +30,12 @@ import {
 import {paginator} from '@google-cloud/paginator';
 import {promisifyAll} from '@google-cloud/promisify';
 import arrify = require('arrify');
-import * as async from 'async';
 import * as extend from 'extend';
 import * as fs from 'fs';
 import * as mime from 'mime-types';
 import * as path from 'path';
+import pLimit from 'p-limit';
```

Comment:
have we opted to move away from `async` to `p-limit` for other libraries. Curious what the advantages are.

Reply:
Good question, this is the only one that I'm aware of. I saw that #731 planned on pulling it in as a regular dependency, so I'm sort of piggy backing off of that.

AFAIK most of the ways we used `async` here can be replaced by native Promises, but managing concurrency is not something that comes out of the box there. In the past I've usually turned to [p-queue](https://www.npmjs.com/package/p-queue), but `p-limit` works here too and we were already slated to use it.

Reply:
Followed the suggestion [here](https://github.com/googleapis/nodejs-spanner/pull/449#issuecomment-442529440)